src/Cores/Storage/src/FileMetadata/Scanners/DepthFirstFolderScanner.cs

```diff
@@ -82,42 +82,55 @@ await foreach (var subFile in RecursiveScanForFilesAsync(folder))
     private async Task EnableFolderWatcherAsync(IMutableFolder folder, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
-        var folderWatcher = await folder.GetFolderWatcherAsync(cancellationToken);
+        try
+        {
+            var folderWatcher = await folder.GetFolderWatcherAsync(cancellationToken);
 
-        cancellationToken.ThrowIfCancellationRequested();
+            cancellationToken.ThrowIfCancellationRequested();
 
-        lock (_knownSubFolders)
+            lock (_knownSubFolders)
+            {
+                if (folder.Id == RootFolder.Id)
+                    _rootFolderWatcher = folderWatcher;
+                else
+                    _knownSubFolders[folder.Id].FolderWatcher = folderWatcher;
+            }
+
+            folderWatcher.CollectionChanged += FolderWatcherOnCollectionChanged;
+        }
+        catch (System.Exception ex)
         {
-            if (folder.Id == RootFolder.Id)
-                _rootFolderWatcher = folderWatcher;
-            else
-                _knownSubFolders[folder.Id].FolderWatcher = folderWatcher;
+            OwlCore.Diagnostics.Logger.LogError("Couldn't get folder watcher", ex);
         }
-
-        folderWatcher.CollectionChanged += FolderWatcherOnCollectionChanged;
     }
 
     private async Task DisableFolderWatcherAsync(IMutableFolder folder, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
+        try
+        {
+            var folderWatcher = folder.Id == RootFolder.Id ? _rootFolderWatcher : _knownSubFolders[folder.Id].FolderWatcher;
```

Comment:
This change was made because folder watching isn't supported on WASM. StorageCore doesn't know about WASM, so we're just using a catch-all for when the folder watcher fails.